src/Analyzers/Core/Analyzers/UseAutoProperty/AbstractUseAutoPropertyAnalyzer.cs

```diff
@@ -251,106 +366,198 @@ bool ShouldAnalyze(SymbolStartAnalysisContext context, INamedTypeSymbol namedTyp
         if (notification.Severity == ReportDiagnostic.Suppress)
             return;
 
-        var getterField = GetGetterField(semanticModel, property.GetMethod, cancellationToken);
-        if (getterField == null)
+        // If the property already contains a `field` expression, then we can't do anything more here.
+        if (SupportsFieldExpression(compilation) && ContainsFieldExpression(propertyDeclaration, cancellationToken))
             return;
 
-        // Only support this for private fields.  It limits the scope of hte program
-        // we have to analyze to make sure this is safe to do.
-        if (getterField.DeclaredAccessibility != Accessibility.Private)
-            return;
+        var getterFields = GetGetterFields(semanticModel, property.GetMethod, fieldNames, cancellationToken);
```

Comment:
this is the crux of the change.  instead of just looking at the proeprty and seeing if it is a simple read/write-through property, we now analyze it to get the *set* of fields that are utilized in both teh getter/setter.  if we find a single viable field that is complimentary with the property and is used in an acceptable fashion in the type, we'll then offer to convert.

note: if it still is a simple read/write-through proeprty, everything works the same as before.  we're just expanding on that for C#13 and above.